### PR TITLE
Migrate to recommended lints

### PR DESCRIPTION
**Pull request by parlough**
Migrates to the `recommended` set of lints from https://pub.dev/packages/lints, removes duplicate linter rules, and fixes the corresponding issues.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 * Removed package list of 2.10 null-safety experiments.
 * Update dependencies to latest.
 * Migration to null-safety.
+* Deprecate `SCREAMING_CAPS` versions of `LicenseNames` constants.
```
Comment (isoos) [inline CHANGELOG.md:7]: Please make it clear that these are breaking changes.

Comment (parlough) [reply CHANGELOG.md]: Sure, how is it now?

**Comment by isoos** [comment]
Enabled CI on the PR, please fix the formatting issue (and if there is anything else).

**Comment by isoos** [comment]
lgtm

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,9 @@
 * Removed package list of 2.10 null-safety experiments.
 * Update dependencies to latest.
 * Migration to null-safety.
+* **BREAKING CHANGES**
+  * Deprecate the `SCREAMING_CAPS` versions of the `LicenseNames` constants. 
+    Use the newly introduced `lowerCamelCase` versions instead.
```
Comment (jonasfj) [inline CHANGELOG.md:9]: How is this breaking? You just deprecated the old names, you didn't remove them.

**Review by jonasfj — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (parlough) [reply CHANGELOG.md:9]: @isoos suggested adding the breaking change descriptor in a previous review. Perhaps due to plans to remove them as part of the release in a follow up or just to make it more clear for those reading the changelog that they need to migrate. 

Let me know if you want me to instead delete the breaking change mention in a new PR.

Comment (isoos) [reply CHANGELOG.md:9]: Yes, this was after we've decided to do a breaking release with pana's null-safety migration. I'll create a PR that removes the old values, and also hides some internals that were exposed as public API.

Comment (parlough) [reply CHANGELOG.md:9]: Sounds great @isoos, thank you!

**Pull request merged**